### PR TITLE
feat(j-s): modifiy ruling

### PR DESCRIPTION
**Pull request by Arnarkari93**
[Asana](https://app.asana.com/0/1199153462262248/1201976501087692/f)

## What

- Add button to modify ruling on signedVerdictOverview
- Update continue and back button when the ruling is being modified
- Pressing the continue button will show the `SigningModal` where the judge has to sign the ruling with an electronic id
- Pressing cancel will navigate back to SignedVerdictOverview 
- Show info pannel if a ruling has been modified on SignedVerdictOverview
- Track the modified history on the backend with a new column `ruling_modified_history` in the database
- Make sure the ruling is sent again to the court service and notifications are sent out with updated messaging

- ## Why

Specify why you need to achieve this

## Screenshots / Gifs

Attach Screenshots / Gifs to help reviewers understand the scope of the pull request

## Checklist:

- [ ] I have performed a self-review of my own code
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] Formatting passes locally with my changes
- [ ] I have rebased against main before asking for a review

### CODE REVIEW

File: apps/judicial-system/backend/src/app/modules/case/case.service.ts [hunk 1]
```diff
@@ -364,20 +364,17 @@ export class CaseService {
...
         address: theCase.prosecutor?.email ?? '',
       },
-      rulingUploadedToS3
-        ? this.formatMessage(m.signedRuling.prosecutorBodyS3, {
-            courtCaseNumber: theCase.courtCaseNumber,
-            courtName: theCase.court?.name?.replace('dómur', 'dómi'),
-            linkStart: `<a href="${this.config.deepLinks.completedCaseOverviewUrl}${theCase.id}">`,
-            linkEnd: '</a>',
-          })
-        : this.formatMessage(m.signedRuling.prosecutorBodyAttachment, {
```
Comment (Arnarkari93) [inline apps/judicial-system/backend/src/app/modules/case/case.service.ts:374]: We got approval for removing this message.

File: apps/judicial-system/web-e2e/src/integration/Prosecutor/InvestigationCases/signedVerdictOverview.spec.ts [hunk 2]
```diff
@@ -0,0 +1,22 @@
+import { SIGNED_VERDICT_OVERVIEW } from '@island.is/judicial-system/consts'
+import { Case } from '@island.is/judicial-system/types'
+
+import { intercept, makeRestrictionCase, makeProsecutor } from '../../../utils'
+
+describe('Signed verdict overview for prosecutors', () => {
```
Comment (oddsson) [inline apps/judicial-system/web-e2e/src/integration/Prosecutor/InvestigationCases/signedVerdictOverview.spec.ts:6]:
```suggestion
describe('Signed verdict overview - Prosecutor - Investigation cases', () => {
```

File: apps/judicial-system/web-e2e/src/integration/Prosecutor/RestrictionCases/signedVerdictOverview.spec.ts [hunk 3]
```diff
@@ -0,0 +1,22 @@
+import { SIGNED_VERDICT_OVERVIEW } from '@island.is/judicial-system/consts'
+import { Case } from '@island.is/judicial-system/types'
+
+import { intercept, makeRestrictionCase, makeProsecutor } from '../../../utils'
+
+describe('Signed verdict overview - prosecutor - restricton cases', () => {
```
Comment (oddsson) [inline apps/judicial-system/web-e2e/src/integration/Prosecutor/RestrictionCases/signedVerdictOverview.spec.ts:6]:
```suggestion
describe('Signed verdict overview - Prosecutor - Restriction cases', () => {
```

File: apps/judicial-system/web/src/routes/Shared/SignedVerdictOverview/SignedVerdictOverviewForm.tsx [hunk 4]
```diff
@@ -398,6 +412,23 @@ const SignedVerdictOverviewForm: React.FC<Props> = (props) => {
                   signatory={workingCase.judge?.name}
                   signingDate={workingCase.rulingDate}
                 />
+                {(user?.role === UserRole.JUDGE ||
+                  user?.role === UserRole.REGISTRAR) && (
```
Comment (Arnarkari93) [inline apps/judicial-system/web/src/routes/Shared/SignedVerdictOverview/SignedVerdictOverviewForm.tsx:416]: Should not only the judge be able to sign the ruling?
```suggestion
                {user?.role === UserRole.JUDGE && (
```

**Pull request merged**